tests/testthat/add.txt

```diff
@@ -0,0 +1,45 @@
+> add_row(tibble(), a = 1)
+<error/tibble_error_incompatible_new_rows>
+Error: New rows can't add columns.
+x Can't find column `a` in `.data`.
+
+> add_row(tibble(), a = 1, b = 2)
+<error/tibble_error_incompatible_new_rows>
+Error: New rows can't add columns.
+x Can't find columns `a` and `b` in `.data`.
+
+> add_row(tibble(), !!!set_names(letters))
+<error/tibble_error_incompatible_new_rows>
+Error: New rows can't add columns.
+x Can't find columns `a`, `b`, `c`, `d`, `e`, and 21 more in `.data`.
+
+> add_row(dplyr::group_by(tibble(a = 1), a))
+<error/tibble_error_add_rows_to_grouped_df>
+Error: Can't add rows to grouped data frames.
+
+> add_row(tibble(a = 1), a = 2, .before = 1, .after = 1)
+<error/tibble_error_both_before_after>
```

Comment:
I'm worried about creating error classes for trivial UI errors. I think it will do more harm than good in the long term, but we'll see.